src/node/gov/handlers/service_state.h

```diff
@@ -0,0 +1,637 @@
+        "/service/javascript-app",
+        HTTP_GET,
+        api_version_adapter(get_javascript_app),
+        no_auth_required)
+      .set_openapi_hidden(true)
+      .install();
+
+    auto get_join_policy = [&](auto& ctx, ApiVersion api_version) {
+      switch (api_version)
+      {
+        case ApiVersion::preview_v1:
+        default:
+        {
+          auto response_body = nlohmann::json::object();
+
+          // Describe SGX join policy
+          {
+            auto sgx_policy = nlohmann::json::object();
+
+            auto sgx_measurements = nlohmann::json::array();
+            auto code_ids_handle =
+              ctx.tx.template ro<ccf::CodeIDs>(ccf::Tables::NODE_CODE_IDS);
+            code_ids_handle->foreach(
+              [&sgx_measurements](
+                const ccf::pal::SgxAttestationMeasurement& measurement,
+                const ccf::CodeStatus& status) {
+                if (status == ccf::CodeStatus::ALLOWED_TO_JOIN)
+                {
+                  sgx_measurements.push_back(measurement.hex_str());
+                }
+                return true;
+              });
+            sgx_policy["measurements"] = sgx_measurements;
+
+            response_body["sgx"] = sgx_policy;
+          }
+
+          // Describe SNP join policy
+          {
+            auto snp_policy = nlohmann::json::object();
+
+            auto snp_measurements = nlohmann::json::array();
+            auto measurements_handle = ctx.tx.template ro<ccf::SnpMeasurements>(
+              ccf::Tables::NODE_SNP_MEASUREMENTS);
+            measurements_handle->foreach(
+              [&snp_measurements](
+                const pal::SnpAttestationMeasurement& measurement,
+                const ccf::CodeStatus& status) {
+                if (status == ccf::CodeStatus::ALLOWED_TO_JOIN)
+                {
+                  snp_measurements.push_back(measurement.hex_str());
+                }
+                return true;
+              });
+            snp_policy["measurements"] = snp_measurements;
+
+            auto snp_host_data = nlohmann::json::object();
+            auto host_data_handle =
+              ctx.tx.template ro<ccf::SnpHostDataMap>(ccf::Tables::HOST_DATA);
+            host_data_handle->foreach(
+              [&snp_host_data](
+                const HostData& host_data, const HostDataMetadata& metadata) {
+                snp_host_data[host_data.hex_str()] = host_data;
+                return true;
+              });
+            snp_policy["hostData"] = snp_host_data;
+
+            auto snp_endorsements = nlohmann::json::object();
+            auto endorsements_handle =
+              ctx.tx.template ro<ccf::SNPUVMEndorsements>(
+                ccf::Tables::NODE_SNP_UVM_ENDORSEMENTS);
+            endorsements_handle->foreach(
+              [&snp_endorsements](
+                const ccf::DID& did,
+                const ccf::FeedToEndorsementsDataMap& feed_info) {
+                snp_endorsements[did] = feed_info;
```

Comment:
Need to update TypeSpec here, as it says this field is a string but we can actually return something more structured.

Reply:
Tracking these in #5691, so resolving here.